zkvm/src/signature/musig.rs

```diff
@@ -0,0 +1,176 @@
+#![allow(non_snake_case)]
+
+use crate::errors::VMError;
+use crate::signature::VerificationKey;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::constants::RISTRETTO_BASEPOINT_POINT;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+
+#[derive(Clone)]
+pub struct PubKey(pub RistrettoPoint);
```

Comment:
maybe use one pubkey type? `VerificationKey` can work, and if we need to make it decompressed, we can make it a struct with private fields for `Option<compressed>` and decompressed point. Compressed is to avoid re-compression cost if instantiating from a compressed point.

Reply:
Done! I will use `VerificationKey` instead of `PubKey` for all instances. For now, it seems fine to pass it around as a `CompressedRistretto` wrapper, can update the struct if it gets to clumsy to wrap and rewrap.